routes/userRouter.js

```diff
@@ -0,0 +1,107 @@
+import express from 'express';
+import Joi from 'joi';
+import ejValidator from 'express-joi-validation';
+
+const userRouter = express.Router();
+const validator = ejValidator.createValidator();
+
+const bodySchema = Joi.object({
+  id: Joi.string().required(),
+  login: Joi.string().alphanum().min(3).max(30).required(),
+  password: Joi.string()
+    .regex(/^[a-zA-Z0-9]{3,30}$/)
+    .required(),
+  isDeleted: Joi.bool().required(),
+  age: Joi.number().integer().min(4).max(130).required(),
+});
+
+let users = [
+  {
+    id: '1',
+    login: 'user1',
+    password: 'password12',
+    age: 26,
+    isDeleted: false,
+  },
+];
+
+userRouter.get('/users/:id', (req, res) => {
+  const { id } = req.params;
+  console.log(users);
+  const user = users.find((user) => user.id === id);
+  if (user) {
+    res.send(user);
+  } else {
+    res.send('User not found');
+  }
+});
+
+userRouter.post('/user', validator.body(bodySchema), (req, res) => {
+  const { body: user } = req;
+  users = [...users, user];
+  res.send('User added successfully');
+});
+
+userRouter.delete('/users/:id', (req, res) => {
+  const { id } = req.params;
+  let userFound = false;
+  const newUsers = users.map((user) => {
+    if (user.id === id) {
+      user.isDeleted = true;
+      userFound = true;
+      return user;
+    }
+    return user;
+  });
+  users = newUsers;
+  if (userFound) {
+    res.send('User deleted successfully');
+  } else {
+    res.send('User not found');
+  }
+});
+
+userRouter.put('/users/:id', validator.body(bodySchema), (req, res) => {
+  const { id } = req.params;
+  const { body: updatedUser } = req;
+  let userFound = false;
+  const newUsers = users.map((user) => {
+    if (user.id === id) {
+      userFound = true;
+      return {
+        ...user,
+        ...updatedUser,
+      };
+    }
+    return user;
+  });
+  users = newUsers;
+  if (userFound) {
+    res.send('User updated successfully');
+  } else {
+    res.send('User not found');
+  }
+});
+
+userRouter.get('/users/getAutoSuggestUsers/:limit', (req, res) => {
+  const { limit } = req.params;
+  const { login } = req.body;
+  const filteredUsers = users
```

Comment:
There are a few problems and suggestions:
1) The route should be /users Method: GET which will return all users. If the person wants to filter further he will implement Step 2 item as elaborated below.
2) Since it is a GET request and GET requests don't send body, we should try to fetch loginSubString and limit as query parameters. You can do that by req.query
eg: /users?loginSubString=user&limit=10